warp-contracts-plugin-quickjs/src/eval/QuickJsEvaluator.ts

```diff
@@ -50,6 +50,42 @@ export class QuickJsEvaluator {
     randomHandle.dispose();
   }
 
+  dummyPromiseEval() {
+    const dummyPromiseEval = this.vm.newFunction('dummyPromise', () => {
+      const promise = this.vm.newPromise();
+      promise.resolve(this.vm.newString(''));
+      promise.settled.then(this.vm.runtime.executePendingJobs);
+      return promise.handle;
+    });
+    this.vm.setProp(this.vm.global, 'dummyPromise', dummyPromiseEval);
+    dummyPromiseEval.dispose();
+  }
+
+  evalExternal() {
+    const readExternalHandle = this.vm.newFunction('readExternal', (processIdHandle, actionHandle) => {
+      const promise = this.vm.newPromise();
+      this.readExternal(processIdHandle, actionHandle).then((result) => {
+        promise.resolve(this.vm.newString(JSON.stringify(result) || ''));
+      });
+      promise.settled.then(this.vm.runtime.executePendingJobs);
+      return promise.handle;
+    });
+    this.vm.setProp(this.vm.global, 'readExternal', readExternalHandle);
+    readExternalHandle.dispose();
+  }
+
+  async readExternal(processIdHandle: QuickJSHandle, actionHandle: QuickJSHandle) {
+    const processId = this.vm.getString(processIdHandle);
+    const action = this.vm.getString(actionHandle);
+    const { dryrun } = await import('@permaweb/aoconnect');
+    const readRes = await dryrun({
+      process: processId,
+      tags: [{ name: 'Action', value: action }],
+      data: '1234'
+    });
+    return JSON.parse(readRes.Messages[0].Data);
```

Comment:
maybe there is no point in parsing the response if we need to stringify it again to pass it to the quickjs..